### PR TITLE
[Merged by Bors] - feat:(Condensed): the functor from `TopCat` to `CondensedSet` is a faithful right adjoint.

### PR DESCRIPTION
**Pull request by dagurtomas**
Also adds some lemmas which improve automation in condensed mathematics.  

---
In #14514 we define compactly generated spaces and show that this functor becomes fully faithful when restricted to them.

- [x] depends on: #14490 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Condensed/TopCatAdjunction.lean [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+universe u
+
+open Condensed CondensedSet CategoryTheory
+
+attribute [local instance] ConcreteCategory.instFunLike
+
+variable (X : CondensedSet.{u})
+
+/-- Auxiliary defiinition to define the topology on `X(*)` for a condensed set `X`. -/
```
Comment (joelriou) [inline Mathlib/Condensed/TopCatAdjunction.lean:24]:
```suggestion
/-- Auxiliary definition to define the topology on `X(*)` for a condensed set `X`. -/
```

File: Mathlib/Condensed/TopCatAdjunction.lean [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+private def _root_.CompHaus.const (S : CompHaus.{u}) (s : S) : CompHaus.of PUnit.{u+1} ⟶ S :=
+  ContinuousMap.const _ s
+
+/-- Auxiliary defiinition to define the topology on `X(*)` for a condensed set `X`. -/
```
Comment (joelriou) [inline Mathlib/Condensed/TopCatAdjunction.lean:28]:
```suggestion
/-- Auxiliary definition to define the topology on `X(*)` for a condensed set `X`. -/
```

**Comment by joelriou** [comment]
Thanks!

bors d+

**Comment by dagurtomas** [comment]
bors merge

**Pull request closed without merging**